contracts/services/neutron-ibc-transfer/README.md

```diff
@@ -0,0 +1,78 @@
+# Valence Neutron IBC Transfer service
+
+The **Valence Neutron IBC Transfer** service allows to transfer funds over IBC from an **input account** on Neutron to an **output account** on a destination chain. It is typically used as part of a **Valence Workflow**. In that context, a **Processor** contract will be the main contract interacting with the Forwarder service.
+
+Note: this service should not be used on another CosmWasm chain than Neutron, which requires some fees to be paid to relayers for IBC transfers. For other CosmWasm chains, prefer using the **Generic IBC Transfer service** instead.
+
+## High-level flow
+
+```mermaid
+---
+title: Neutron IBC Transfer Service
+---
+graph LR
+  IA((Input
+      Account))
+  OA((Output
+		  Account))
+  P[Processor]
+  S[Neutron IBC Transfer
+    Service]
+  subgraph Neutron
+  P -- 1/IbcTransfer --> S
+  S -- 2/Query balances --> IA
+  S -- 3/Do Send funds --> IA
+  end
+  subgraph Chain 2
+  IA -- 4/IBC Transfer --> OA
+  end
+```
+
+## Configuration
+
+The service is configured on instantiation via the `ServiceConfig` type.
+```rust
+struct ServiceConfig {
+  // Account from which the funds are pulled (on the source chain)
+  input_addr: ServiceAccountType,
+  // Account to which the funds are sent (on the destination chain)
+  output_addr: String,
+  // Denom of the token to transfer
+  denom: UncheckedDenom,
+  // Amount to be transferred, either a fixed amount or the whole available balance.
+  amount: IbcTransferAmount,
+  // Memo to be passed in the IBC transfer message.
+  memo: String,
+  // Information about the destination chain.
+  remote_chain_info: RemoteChainInfo,
+  // Denom map for the Packet-Forwarding Middleware, to perform a multi-hop transfer.
+  denom_to_pfm_map: BTreeMap<String, PacketForwardMiddlewareConfig>,
+}
+
+// Defines the amount to be transferred, either a fixed amount or the whole available balance.
+enum IbcTransferAmount {
+  // Transfer the full available balance of the input account.
+  FullAmount,
+  // Transfer the specified amount of tokens.
+  FixedAmount(Uint128),
+}
+
+pub struct RemoteChainInfo {
+  // Channel of the IBC connection to be used.
+  channel_id: String,
+  // Port of  the IBC connection to be used.
+  port_id: Option<String>,
```

Comment:
Similar to generic contract but one difference: here it's actually being used in the contract but are we even contemplating doing something different to ibc transfers? Looking at the contract we are using an MsgTransfer on both contracts so anything that is not "transfer" would not be valid.

I could see this making sense if we would have different kinds of IBC messaging supported here (e.g. IBC to another contract), but I think we would split that into a different service and using a base account with  ibc logic implemented (and therefore ibc ports, callbacks...)

Reply:
I thought the we had a case for Covenant where we were IBC transferring CW20 tokens .. but the incoming port was `transfer` so it's still not needed in that case.